index.js

```diff
@@ -21,10 +21,6 @@ module.exports = user => {
 		username: res.body.user.username,
 		website: res.body.user.external_url || ''
 	})).catch(err => {
-		if (err.statusCode === 404) {
-			return 'Not found!';
-		} else {
-			return 'Error!'
-		}
+		return err;
```

Comment:
I think what you had was a good start. But like I said, I believe it would be better to return undefined if the user couldn't be found. Just returning the error makes error handling even harder.

```js
if (err.statusCode === 404) {
    return undefined;
}

throw err;
```

But maybe let's wait the feedback from @kevva before we continue. Not sure if handling those things here actually makes sense. It's up to the developer that uses this library to gracefully handle errors if something goes wrong and thus to implement the catch handler.